network/transparent_vlan_endpointclient_linux.go

```diff
@@ -126,32 +125,94 @@ func (client *TransparentVlanEndpointClient) AddEndpoints(epInfo *EndpointInfo)
 	})
 }
 
-func (client *TransparentVlanEndpointClient) createNetworkNamespace(vmNS, numRetries int) error {
-	var isNamespaceUnique bool
+// Called from PopulateVM, Namespace: VM and Vnet
+func (client *TransparentVlanEndpointClient) ensureCleanPopulateVM() error {
+	// Clean up vlan interface in the VM namespace and ensure the network namespace (if it exists) has a vlan interface
+	logger.Info("Checking if NS and vlan veth exists...")
+	var existingErr error
+	client.vnetNSFileDescriptor, existingErr = client.netnsClient.GetFromName(client.vnetNSName)
+	if existingErr == nil {
+		// The namespace exists
+		vlanIfErr := ExecuteInNS(client.nsClient, client.vnetNSName, func() error {
+			// Ensure the vlan interface exists in the namespace
+			_, err := client.netioshim.GetNetworkInterfaceByName(client.vlanIfName)
+			return errors.Wrap(err, "failed to get vlan interface in namespace")
+		})
+
+		// If the vlan veth for sure doesn't exist, we delete the vnet ns, but if we can't query, we can't delete the ns (the vlan veth might exist!)
+		if vlanIfErr != nil {
+			if strings.Contains(vlanIfErr.Error(), "no such network interface") {
+				logger.Info("Vlan interface doesn't exist even though network namespace exists, deleting network namespace...")
+				delErr := client.netnsClient.DeleteNamed(client.vnetNSName)
+				if delErr != nil {
+					return errors.Wrap(delErr, "failed to cleanup/delete ns after noticing vlan veth does not exist")
+				}
+			} else {
+				return errors.Wrap(vlanIfErr, "could not determine if vlan veth exists in vnet namespace")
+			}
+		}
+	} else {
```

Comment:
i dont think log is required.. anyway we log in populateVM right?

Reply:
Yes we do log in populateVM, but here (see above comment) we want to know if GetNetworkInterfaceByName returned an error other than "interface not found", in case the method itself fails. Again, if you've only ever seen it fail if the interface is not found, we can probably remove the log.

Reply:
ya lets remove the log.. anyway we log in populateVM